core/src/main/java/com/sequenceiq/cloudbreak/rotation/executor/CustomJobExecutor.java

```diff
@@ -0,0 +1,39 @@
+package com.sequenceiq.cloudbreak.rotation.executor;
+
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.rotation.context.CustomJobRotationContext;
+import com.sequenceiq.cloudbreak.rotation.secret.RotationExecutor;
+import com.sequenceiq.cloudbreak.rotation.secret.step.CommonSecretRotationStep;
+import com.sequenceiq.cloudbreak.rotation.secret.step.SecretRotationStep;
+
+@Component
+public class CustomJobExecutor implements RotationExecutor<CustomJobRotationContext> {
+
+    @Override
+    public void rotate(CustomJobRotationContext rotationContext) throws Exception {
+        Optional.ofNullable(rotationContext.getRotationJob()).stream().forEach(Runnable::run);
+    }
+
+    @Override
+    public void rollback(CustomJobRotationContext rotationContext) throws Exception {
+        rotationContext.getRollbackJob().stream().forEach(Runnable::run);
+    }
+
+    @Override
+    public void finalize(CustomJobRotationContext rotationContext) throws Exception {
+        rotationContext.getFinalizeJob().stream().forEach(Runnable::run);
+    }
```

Comment:
there is only one runnable here for every step, I do not get why do we need foreach here

Reply:
I see that optional has stream, though it is strange, it has only one or zero value always, for me foreach is confusing but we can keep this since it makes sense, there will be no multiple items

Reply:
Functional idiom for every "container" type. In this case it's used instead of null check or isPresent check